commons/src/test/java/io/aiven/kafka/tiered/storage/commons/manifest/SegmentManifestV1SerdeTest.java

```diff
@@ -0,0 +1,115 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.manifest;
+
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Base64;
+
+import io.aiven.kafka.tiered.storage.commons.RsaKeyAwareTest;
+import io.aiven.kafka.tiered.storage.commons.chunkindex.FixedSizeChunkIndex;
+import io.aiven.kafka.tiered.storage.commons.manifest.serde.SecretKeyDeserializer;
+import io.aiven.kafka.tiered.storage.commons.manifest.serde.SecretKeySerializer;
+import io.aiven.kafka.tiered.storage.commons.security.EncryptionKeyProvider;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SegmentManifestV1SerdeTest extends RsaKeyAwareTest {
+    static final FixedSizeChunkIndex INDEX =
+        new FixedSizeChunkIndex(100, 1000, 110, 110);
+    static final SecretKey SECRET_KEY = new SecretKeySpec(new byte[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, "AES");
+    static final byte[] AAD = {10, 11, 12, 13};
+
+    static final String WITH_ENCRYPTION_WITHOUT_SECRET_KEY_JSON =
+        "{\"version\":\"1\","
+            + "\"chunkIndex\":{\"type\":\"fixed\",\"originalChunkSize\":100,"
+            + "\"originalFileSize\":1000,\"transformedChunkSize\":110,\"finalTransformedChunkSize\":110},"
+            + "\"compression\":false,\"encryption\":{\"aad\":\"CgsMDQ==\"}}";
+    static final String WITHOUT_ENCRYPTION_JSON =
+        "{\"version\":\"1\","
+            + "\"chunkIndex\":{\"type\":\"fixed\",\"originalChunkSize\":100,"
+            + "\"originalFileSize\":1000,\"transformedChunkSize\":110,\"finalTransformedChunkSize\":110},"
+            + "\"compression\":false}";
+
+    ObjectMapper mapper;
+    EncryptionKeyProvider encryptionKeyProvider;
+
+    @BeforeEach
+    void init() throws IOException {
+        mapper = new ObjectMapper();
+        mapper.registerModule(new Jdk8Module());
+
+        try (final FileInputStream publicKeyFis = new FileInputStream(publicKeyPem.toFile());
+             final FileInputStream privateKeyFis = new FileInputStream(privateKeyPem.toFile())) {
```

Comment:
Should we stick to `Files.newInputStream()` as in other tests? May be confusing to have both across tests

Reply:
Fixed